### DISCUSSION TITLE
How to contribute standard cells ?

### DISCUSSION THREAD

**Discussion by Blebowski** [IHP-GmbH/IHP-Open-PDK#152] [General]
Hi,

If I wanted to contribute a standard cell, is there a way how to do it with OS tools?
I have not seen XSchem schematics for the IHP standard cells in this repository,
nor TBs for the standard cell verification.

Is it correct to assume the standard cell design (schematic, layout, characterization into .lib)
was done using commercial toolchain ?

**Comment by sergeiandreyev** [comment]
Hi @Blebowski, you're right that current list of std cells was generated using commercial tools
But we're open to contributions and you can generate the stdcells using OS EDA - tools to create schematics/testbenches, simulation models, DRC/LVS are available
We have even a list of additional cells that was planned but wasn't finished during last year Hackathon in ETHZ
Please note that there will be an internal review of the cells developed by the open source community

**Comment by Blebowski** [comment]
Hi,
thanks for a quick reply. I am aware of the open-source schem + sim, layout, DRC / LVS tools,
but I don't know if something exists that can be used to run characterization (create `.lib` file).

Are you aware of something like that ?

**Comment by sergeiandreyev** [comment]
for the Liberty generation there is this project called 'Librecell': https://github.com/Ravenslofty/librecell/tree/anneal/librecell-lib
but it's rather old, we did not test it and the support is unknown
certainly, if you want to generate full set of EDA views for stdcells there are some pieces missing (maybe we will be working towards filling these gaps), but still you can just contribute with layout/schematic and we will generate the remaining views internally

**Reply by Blebowski** [reply]
Thank you I did not known the project.

**Comment by sergeiandreyev** [comment]
> I have not seen XSchem schematics for the IHP standard cells in this repository

We will provide some sample xschem schematic views for combinational and sequential cells to take as an example for the open source community contributions

**Comment by Blebowski** [comment]
Hi,
I recently ran accross:
https://github.com/stineje/CharLib

I have not tried it yet, but It seems to be more maintained than librecell

**Comment by sergeiandreyev** [comment]
Now there are other projects as well:
https://github.com/snishizawa/libretto
https://github.com/Blebowski/OpenChar
